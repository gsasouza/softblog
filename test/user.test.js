const request = require('supertest-as-promised');
const app = require('../server');


describe('User', () => {
  describe('GET => /users', () => {
    it('Should return an array with the users', async () => {
      const res = await request(app).get('/api/users');
      expect(res.statusCode).toBe(200);
      expect(res.body).toMatchSnapshot();
    });
  });

  describe('POST => /users', () => {
    it('Should create a user', async () => {
      const res = await request(app)
        .post('/api/users')
        .send({
          username: 'joaozinho',
          password: '1234',
          name: 'JoÃ£o',
          type: 'ADMIN'
        });
      expect(res.statusCode).toBe(201);
      expect(res.body).toMatchSnapshot();
    });
  });

  describe('GET => /users/{username}', () => {
    it('Should get a specific user', async () => {
      const res = await request(app).get('/api/users/joaozinho');
      expect(res.statusCode).toBe(200);
      expect(res.body).toMatchSnapshot();
    });
  });

  describe('PUT => /users/{username}', () => {
    it('Should update a specific user', async () => {
      const res = await request(app)
        .put('/api/users/joaozinho')
        .send({ name: 'Little John' });
      expect(res.statusCode).toBe(200);
      expect(res.body).toMatchSnapshot();
    });
  });

  describe('DELETE => /users/{username}', () => {
    it('Should delete a specific user', async () => {
      const res = await request(app).delete('/api/users/joaozinho');
      expect(res.statusCode).toBe(200);
      expect(res.body).toMatchSnapshot();
    });
  });
});
