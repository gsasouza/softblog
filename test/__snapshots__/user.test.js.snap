// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User DELETE => /users/{username} Should delete a specific user 1`] = `Object {}`;

exports[`User GET => /users Should return an array with the users 1`] = `Array []`;

exports[`User GET => /users/{username} Should get a specific user 1`] = `
Object {
  "__v": 0,
  "name": "JoÃ£o",
  "type": "ADMIN",
  "username": "JOAOZINHO",
}
`;

exports[`User POST => /users Should create a user 1`] = `Object {}`;

exports[`User PUT => /users/{username} Should update a specific user 1`] = `Object {}`;
